### PR TITLE
docs: update cohere documentation

### PR DESCRIPTION
**Pull request by sepiatone**
**Description:** Update of Cohere documentation (main provider page)
**Issue:** After addition of the Cohere partner package, the documentation was out of date
**Dependencies:** None

### CODE REVIEW

File: docs/docs/integrations/providers/cohere.mdx [hunk 1]
```diff
@@ -4,62 +4,74 @@
...
+|Rerank Retriever|Rank strings based on relevance|[rerank](https://docs.cohere.com/reference/rerank)|
+
 
 ## Quick copy examples
 
 ### Chat
 
 ```python
-from langchain_community.chat_models import ChatCohere
+from langchain_cohere.chat_model import ChatCohere
```
Comment (ccurme) [inline docs/docs/integrations/providers/cohere.mdx:30]:
```suggestion
from langchain_cohere.chat_models import ChatCohere
```

File: docs/docs/integrations/providers/cohere.mdx [hunk 2]
```diff
@@ -4,62 +4,74 @@
...
+Usage of the Cohere [chat model](/docs/integrations/chat/cohere)
+
 ### LLM
 
 
 ```python
-from langchain_community.llms import Cohere
+from langchain_cohere.llms import Cohere
 
 llm = Cohere(model="command")
```
Comment (ccurme) [inline docs/docs/integrations/providers/cohere.mdx:45]:
```suggestion
llm = Cohere()
```
Interestingly if model is specified here, I get 
`TypeError: langchain_cohere.llms.completion_with_retry() got multiple values for keyword argument 'model'`
below when calling `llm.invoke`.

File: docs/docs/integrations/providers/cohere.mdx [hunk 3]
```diff
@@ -4,62 +4,74 @@
...
-|API|description|Endpoint docs|Import|Example usage|
-|---|---|---|---|---|
-|Chat|Build chat bots|[chat](https://docs.cohere.com/reference/chat)|`from langchain_community.chat_models import ChatCohere`|[cohere.ipynb](/docs/integrations/chat/cohere)|
-|LLM|Generate text|[generate](https://docs.cohere.com/reference/generate)|`from langchain_community.llms import Cohere`|[cohere.ipynb](/docs/integrations/llms/cohere)|
-|RAG Retriever|Connect to external data sources|[chat + rag](https://docs.cohere.com/reference/chat)|`from langchain.retrievers import CohereRagRetriever`|[cohere.ipynb](/docs/integrations/retrievers/cohere)|
-|Text Embedding|Embed strings to vectors|[embed](https://docs.cohere.com/reference/embed)|`from langchain_community.embeddings import CohereEmbeddings`|[cohere.ipynb](/docs/integrations/text_embedding/cohere)|
-|Rerank Retriever|Rank strings based on relevance|[rerank](https://docs.cohere.com/reference/rerank)|`from langchain.retrievers.document_compressors import CohereRerank`|[cohere.ipynb](/docs/integrations/retrievers/cohere-reranker)|
+|API|description|Endpoint docs|
+|---|---|---|
+|Chat|Build chat bots|[chat](https://docs.cohere.com/reference/chat)|
```
Comment (ccurme) [inline docs/docs/integrations/providers/cohere.mdx:18]: Why drop the example imports?

File: docs/docs/integrations/providers/cohere.mdx (hunk 2 above)
Comment (sepiatone) [reply docs/docs/integrations/providers/cohere.mdx]: yes, you're right

File: docs/docs/integrations/providers/cohere.mdx (hunk 3 above)
Comment (sepiatone) [reply docs/docs/integrations/providers/cohere.mdx:18]: I did not think it added any extra information - the code examples below for each component have it.

if you feel that it is helpful, I'll add it back. (in general I agree that more information in the documentation is better even if it is repeated)

File: docs/docs/integrations/providers/cohere.mdx [hunk 4]
```diff
@@ -4,62 +4,74 @@
...
+llm = Cohere()
 print(llm.invoke("Come up with a pet name"))
 ```
 
+Usage of the Cohere (legacy) [LLM model](/docs/integrations/llms/cohere) 
 
 ### RAG Retriever
 
 ```python
 from langchain_community.chat_models import ChatCohere
```
Comment (ccurme) [inline docs/docs/integrations/providers/cohere.mdx:54]:
```suggestion
from langchain_cohere.chat_models import ChatCohere
```

**Review by ccurme — APPROVED** [review]

**Comment by sepiatone** [comment]
@ccurme i've taken in all your review comments

**Review by ccurme — APPROVED** [review]

**Pull request merged**